Sources/ArcGISToolkit/Components/Form/TextEntryFooter.swift

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FormsPlugin
+import SwiftUI
+
+/// A view shown at the bottom of each text entry element in a form.
+struct TextEntryFooter: View {
+    /// <#Description#>
+    @State private var validationError: LengthError? = nil
+    
+    /// The current length of the text in the text entry field.
+    let currentLength: Int
+    
+    /// A Boolean value indicating whether the text entry field is focused.
+    let isFocused: Bool
+    
+    /// The description of the text entry field.
+    let description: String
+    
+    /// A Boolean value indicating whether the text entry field is required.
+    let isRequired: Bool
+    
+    /// The maximum allowable length of text in the text entry field.
+    let maxLength: Int
+    
+    /// The minimum allowable length of text in the text entry field.
+    let minLength: Int
+    
+    var body: some View {
+        HStack(alignment: .top) {
+            if let validationError {
+                switch validationError {
+                case .emptyWhenRequired:
+                    requiredText
+                case .tooLong:
+                    maximumText
+                case .tooShort:
+                    minimumText
+                }
+            } else if !description.isEmpty {
+                Text(description)
+            } else if description.isEmpty && isFocused {
```

Comment:
don't need to check `description.isEmpty` as that was just checked in the previous `if`.
```suggestion
            } else if isFocused {
```

Reply:
The design from Field Maps has the concept of "never focused", and doesn't show the constraint text until it has had the keyboard focus. I don't see that taken into account here.

Reply:
Good point, I think latest commit should take this into account.

"Character Constraint" seems a bit ambiguous in the design and is the difference between the two places where I see "Never Focused" come into play.

<img width="675" alt="Screenshot 2023-06-30 at 17 28 27" src="https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/16397058/e9729bdd-8fa6-4ea9-adfd-93ce7eb1b3e1">
<img width="676" alt="Screenshot 2023-06-30 at 17 28 38" src="https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/16397058/44bb1fee-27c5-4f9b-b8fb-af82911c52e1">

Because min and max length are both non-optional, it seems that there is always a character constraint and so I'm now interpreting "Character Constraint" to only apply when text field has never reached it's minimum and never again after that.

Reply:
Ok. Thanks.